### ISSUE TITLE
Gaps when creating stairs with walls

### ISSUE THREAD

**Issue by MCrafterzz** [ranjian0/building_tool#112]
**Describe the bug**
Gaps

**To Reproduce**
Steps to reproduce the behaviour:
1. Create stairs
2. Select wall railing type

**Expected behaviour**
Walls not creating gaps

**Screenshots**
![image](https://user-images.githubusercontent.com/19669673/83774965-b1f9fb80-a686-11ea-923d-baf9a23850bf.png)

**Comment by ranjian0** [comment]
@luckykadam, what are your thoughts on this?

**Comment by luckykadam** [comment]
The cause of these issues is the common Railing modules, because it is not possible to satisfy completely both Stairs and Balcony.
One solution is to have separate railing modules for both, but this method will cause code duplication, inconsistent behaviour, hard to maintain.
I believe better solution is to share the Railing module and have corner cases handled in Stair and Baclony.

Talking about specific issues:
1. gaps in wall fill type is a known issue. Stairs module should take care of this.
2. short railing posts are caused by wrong use of density. What do you expect by setting number of posts less than number of stair steps? The default value produces correct behaviour. If you change the value, you should be aware of what it does. **Not an issue according to me**

**Comment by ranjian0** [comment]
For no. 2 I still strongly feel that railing posts should not touch the ground.

![Imgur](https://i.imgur.com/ADQu10g.jpg)

This would keep density option for posts consistent across all usages. Having two unrelated properties that depend on each other for correct results is not intuitive. Also, note that keeping the number of stair steps and posts equal is not obvious as railing posts rely on a density option, not count.

**Comment by MCrafterzz** [comment]
> For no. 2 I still strongly feel that railing posts should not touch the ground.
> 
> ![Imgur](https://camo.githubusercontent.com/7379126c836b7212a6f2731b349a93fb6a1c1f7f/68747470733a2f2f692e696d6775722e636f6d2f414451753130672e6a7067)
> 
> This would keep density option for posts consistent across all usages. Having two unrelated properties that depend on each other for correct results is not intuitive. Also, note that keeping the number of stair steps and posts equal is not obvious as railing posts rely on a density option, not count.

Is there a reason why they don't depend on count? You could have it for every nth step and then have a offset ehich determines where on the step the posts are. This would be much more intuitive in my opinion.

**Comment by luckykadam** [comment]
> Is there a reason why they don't depend on count?

Yes, there is a reason. Railing settings are not aware of Stairs or Balcony, and steps is a concept only in Stairs. I encourage you to come up with an alternative design.

> For no. 2 I still strongly feel that railing posts should not touch the ground.

Agreed! This is one type of railing posts, and we should plan to support this. But, we cannot ignore the type where it touched the ground. This will also pave path for selecting custom objects for post.

![Rapide-r-ponse-sur-enqu-te-cristal](https://user-images.githubusercontent.com/11018027/83940420-936c3f80-a801-11ea-99a0-d6cce674d2c8.jpg)

**Comment by ranjian0** [comment]
@MCrafterzz 
Railing module is also used by balcony, adding a dependency to stairs would break modularity and complicate the code.

**Comment by MCrafterzz** [comment]
> @MCrafterzz
> Railing module is also used by balcony, adding a dependency to stairs would break modularity and complicate the code.

Can't it raycast towards the ground then somehow to find the hight of the step where it's created?

**Comment by ranjian0** [comment]
> > @MCrafterzz
> > Railing module is also used by balcony, adding a dependency to stairs would break modularity and complicate the code.
> 
> Can't it raycast towards the ground then somehow to find the hight of the step where it's created? Then just move the post's bottom edges to that position. There's probably a way to do this in blender.

From your image, you should see that the posts are not quite in the appropriate positions. Some are floating and others are inside the stairs.
